### PR TITLE
Pre sentence report context API endpoint

### PR DESCRIPTION
**Pull request by stevomcallister**

### CODE REVIEW

File: .idea/runConfigurations/pre_sentence_reports_to_delius.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <component name="ProjectRunConfigurationManager">
   <configuration default="false" name="pre-sentence-reports-to-delius" type="SpringBootApplicationConfigurationType" factoryName="Spring Boot">
-    <option name="ACTIVE_PROFILES" value="dev" />
+    <option name="ACTIVE_PROFILES" value="oracle" />
```
Comment (marcus-bcl) [inline .idea/runConfigurations/pre_sentence_reports_to_delius.xml:3]: Did you mean to commit this?

File: projects/pre-sentence-reports-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/courtreport/CourtReportRepository.kt [hunk 2]
```diff
@@ -1,5 +1,64 @@
...
+         JOIN EVENT e ON ca.event_id = e.event_id
+         JOIN COURT c ON c.COURT_ID = ca.COURT_ID
+         JOIN PROBATION_AREA lja ON lja.PROBATION_AREA_ID = c.PROBATION_AREA_ID
+         JOIN MAIN_OFFENCE mo ON mo.EVENT_ID = e.EVENT_ID
+         LEFT OUTER JOIN R_OFFENCE moo ON moo.OFFENCE_ID = mo.OFFENCE_ID
+         LEFT OUTER JOIN ADDITIONAL_OFFENCE ao ON ao.EVENT_ID = e.EVENT_ID
+         LEFT OUTER JOIN OFFENDER_ADDRESS a ON a.OFFENDER_ID = o.OFFENDER_ID
+         LEFT OUTER JOIN R_STANDARD_REFERENCE_LIST ast ON ast.STANDARD_REFERENCE_LIST_ID = a.ADDRESS_STATUS_ID
+WHERE 1=1 
+  AND cr.COURT_REPORT_ID = :reportId
```
Comment (marcus-bcl) [inline projects/pre-sentence-reports-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/courtreport/CourtReportRepository.kt:53]: This should be using the `DOCUMENT.EXTERNAL_REFERENCE` column, the pre-sentence service doesn't know the `COURT_REPORT_ID`.

File: projects/pre-sentence-reports-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/presentencereport/PreSentenceReportService.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+@Service
+class PreSentenceReportService(
+    private val courtReportRepository: CourtReportRepository,
+    private val objectMapper: ObjectMapper
+) {
+    fun getPreSentenceReportContext(reportId: String): PreSentenceReportContext {
+        val json = courtReportRepository.getCourtReportContextJson(reportId)
+            ?: throw NotFoundException("CourtReport", "id", reportId)
+        return objectMapper.readValue(json, PreSentenceReportContext::class.java)
```
Comment (marcus-bcl) [inline projects/pre-sentence-reports-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/presentencereport/PreSentenceReportService.kt:16]: Are we definitely happy with this approach?  i.e. creating JSON in the database, deserializing into a PreSentenceReportContext object, then Spring re-serializing it to send in the response?

File: .idea/runConfigurations/pre_sentence_reports_to_delius.xml (hunk 1 above)
Comment (stevomcallister) [reply .idea/runConfigurations/pre_sentence_reports_to_delius.xml:3]: nope

File: projects/pre-sentence-reports-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/courtreport/CourtReportRepository.kt (hunk 2 above)
Comment (stevomcallister) [reply projects/pre-sentence-reports-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/courtreport/CourtReportRepository.kt:53]: ahh ok

File: projects/pre-sentence-reports-to-delius/deploy/values-dev.yml [hunk 4]
```diff
@@ -4,6 +4,9 @@ env:
   SENTRY_ENVIRONMENT: dev
   SPRING_JMS_TEMPLATE_DEFAULT-DESTINATION: Digital-Prison-Services-dev-pre_sentence_report_hmpps_queue
   SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_HMPPS-AUTH_TOKEN-URI: https://sign-in-dev.hmpps.service.justice.gov.uk/auth/oauth/token
+  SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_JWK_SET_URI: "https://sign-in-dev.hmpps.service.justice.gov.uk/auth/.well-known/jwks.json"
+  SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_ISSUER_URI: "https://sign-in-dev.hmpps.service.justice.gov.uk/auth/issuer"
```
Comment (marcus-bcl) [inline projects/pre-sentence-reports-to-delius/deploy/values-dev.yml:8]: What does this do out of interest?  Usually I've only ever configured the JWKS URI.

**Review by marcus-bcl — APPROVED** [review]
Looks good.  Could we make sure the PR title / commit message on main ends up as something more meaningful than "initial commit"? 😄

**Pull request merged**